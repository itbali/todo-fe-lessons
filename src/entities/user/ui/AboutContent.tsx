import {forwardRef, useState} from 'react';
import Typography from "@mui/material/Typography";
import CustomButton from "../../../shared/ui/kit/CustomButton.tsx";

const AboutContent = forwardRef<HTMLSpanElement | null>((_props,ref) => {
    const [isShow, setIsShow] = useState(false);

    return (
        <>
            <Typography variant={'h1'} ref={ref}>
                About
            </Typography>
            <CustomButton onClick={()=>setIsShow(!isShow)}>show</CustomButton>
            {isShow &&
                <Typography variant={'body1'}>
                    This is simple ToDo application
                </Typography>
            }
        </>
    );
});

export default AboutContent;