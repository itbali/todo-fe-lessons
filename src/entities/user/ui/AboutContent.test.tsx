import {describe, expect, it} from "vitest";
import {render, screen, fireEvent, act} from '@testing-library/react'
import AboutContent from "./AboutContent.tsx"

describe('About content page',()=>{
    it('should render component', ()=>{
        const rendered = render(<AboutContent />)
        expect(rendered.baseElement).toMatchSnapshot()
        // expect(screen.getByText('About')).toBeDefined()
        // const button = screen.getByText('show')
        //
        // expect(button).toBeDefined()
        //
        // act(()=>{
        //     fireEvent.click(button)
        // })
        //
        // expect(screen.getByText('This is simple ToDo application')).toBeDefined()
    })
})