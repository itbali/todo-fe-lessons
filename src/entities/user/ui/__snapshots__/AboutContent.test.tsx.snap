// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About content page > should render component 1`] = `
<body>
  <div>
    <h1
      class="MuiTypography-root MuiTypography-h1 css-js3oe9-MuiTypography-root"
    >
      About
    </h1>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1yujjup-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span>
        show
      </span>
    </button>
  </div>
</body>
`;
